evalml/pipelines/components/transformers/preprocessing/transform_primitive_components.py

```diff
@@ -0,0 +1,154 @@
+        super().__init__(random_seed=random_seed, **kwargs)
+
+    @property
+    @classmethod
+    @abstractmethod
+    def _transform_primitives(cls):
+        """Return the transform primitives extracted from this component."""
+
+    @abstractmethod
+    def _get_columns_to_transform(self, X):
+        """Return the columns that the primitives will transform."""
+
+    @abstractmethod
+    def _get_feature_types_for_featuretools(self, X):
+        """Get a mapping from column name to the feature tools type.
+
+        This is needed for dfs. Hopefully, once the ww/ft integration is complete this will be redundant.
+        """
+
+    def _make_entity_set(self, X):
+        X_to_transform = X[self._columns]
+
+        # featuretools expects str-type column names
+        X_to_transform.rename(columns=str, inplace=True)
+        ft_variable_types = self._get_feature_types_for_featuretools(X)
+        es = ft.EntitySet()
+        es.entity_from_dataframe(
+            entity_id="X",
+            dataframe=X_to_transform,
+            index="index",
+            make_index=True,
+            variable_types=ft_variable_types,
+        )
+        return es
+
+    def fit(self, X, y=None):
+        X = infer_feature_types(X)
+        self._columns = self._get_columns_to_transform(X)
+        if len(self._columns) == 0:
+            return self
+
+        es = self._make_entity_set(X)
+        self._features = ft.dfs(
+            entityset=es,
+            target_entity="X",
+            trans_primitives=self._transform_primitives,
+            max_depth=1,
+            features_only=True,
+        )
+        return self
+
+    def transform(self, X, y=None):
+        X_ww = infer_feature_types(X)
+        if self._features is None or len(self._features) == 0:
+            return X_ww
+
+        es = self._make_entity_set(X_ww)
+        features = ft.calculate_feature_matrix(features=self._features, entityset=es)
+
+        features.set_index(X_ww.index, inplace=True)
+
+        X_ww = X_ww.ww.drop(self._columns)
+        for col in features:
+            X_ww.ww[col] = features[col]
+
+        all_created_columns = self._get_feature_provenance().values()
+        to_categorical = {
+            col: "Categorical"
+            for feature_list in all_created_columns
+            for col in feature_list
+        }
+        X_ww.ww.set_types(to_categorical)
+        return X_ww
+
+    @staticmethod
+    def _get_primitives_provenance(features):
+        provenance = {}
+        for feature in features:
+            input_col = feature.base_features[0].get_name()
+            # Return a copy because `get_feature_names` returns a reference to the names
+            output_features = [name for name in feature.get_feature_names()]
+            if input_col not in provenance:
+                provenance[input_col] = output_features
+            else:
+                provenance[input_col] += output_features
+        return provenance
```

Comment:
Shame this is different from the component graph provenance, but nothing to do there.